src/spikeinterface/extractors/whitematterrecordingextractor.py

```diff
@@ -0,0 +1,78 @@
+from pathlib import Path
+from typing import List, Union, Optional
+
+from spikeinterface.core import BinaryRecordingExtractor
+from spikeinterface.core.core_tools import define_function_from_class
+
+
+class WhiteMatterRecordingExtractor(BinaryRecordingExtractor):
+    """
+    RecordingExtractor for the WhiteMatter binary format.
+
+    The recording format is a raw binary file containing int16 data,
+    with an 8-byte header offset.
+
+    Parameters
+    ----------
+    file_paths : list or Path
+        List of paths to the binary files.
+    sampling_frequency : float
+        The sampling frequency.
+    num_channels : int
+        Number of channels in the recording.
+    channel_ids : list or None, default: None
+        A list of channel ids. If None, channel_ids = list(range(num_channels)).
+    time_axis : int, default: 0
```

Comment:
I don't think`time_axis` should be an argument here as this is fxied right?
Also, why provide more than one file_path, I think the data that we had is mono-segment, right?

Reply:
Yeah, I guess that the point of having this extractor is to simplify the interface and fix the things that we can fix (like you are doing with offset).

Why we would not fix these two if we know that the time axis and the number of segments?

Reply:
I think we should fix them (see latest), just didn't consider it that deeply.